[cpp-httplib] 使用clang安装cpp-httplib失败

### Xmake 版本

v3.0.8

### 操作系统版本和架构

Win11

### 描述问题

使用clang安装cpp-httplib失败

### 期待的结果

正常安装

### 工程配置

```lua
add_rules("mode.debug", "mode.release")

set_toolchains("clang")
set_languages("c++23")

add_requires("cpp-httplib[ssl]")

target("cpp-httplib-issue")
    add_packages("cpp-httplib")
    add_files("src/*.cpp")
```

### 附加信息和错误日志

```console
➜ xmake f -cv
checking for platform ... windows (x64)
checking for cl.exe ... D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\bin\HostX64\x64\cl.exe
checking for Microsoft C/C++ Compiler (x64) ... ok
checking for Microsoft Visual Studio (x64) version ... 2026
checking for zig ... no
checking for nim ... no
checking for dotnet ... C:\Program Files\dotnet\dotnet.exe
checking for .NET SDK version ... 8.0.419
checking for .NET SDK directory ... C:\Program Files\dotnet\sdk\8.0.419
checking for git ... D:\scoop\apps\git\current\cmd\git.exe
checking for gzip ... no
checking for 7z ... D:\scoop\apps\xmake\current\winenv\bin\7z
D:\scoop\apps\git\current\cmd\git.exe rev-parse HEAD
checking for clang ... D:\scoop\apps\llvm\current\bin\clang
checking for LLVM Clang C/C++ Compiler (x64) ... ok
checking for Microsoft Visual Studio (x64) version ... 2026
checking for ninja ... no
checking for ninja ... no
checking for ninja ... D:\scoop\shims\ninja.exe
checking for cmake ... no
checking for cmake ... no
checking for cmake ... D:\scoop\shims\cmake.exe
checking for nasm ... no
checking for perl ... no
checking for jom.exe ... no
checking for xmake::openssl3 ... openssl3 3.6.1
checking for xmake::cpp-httplib ... no
checking for vcpkg.exe ... D:\scoop\apps\vcpkg\current\vcpkg.exe
checking for clang ... D:\scoop\apps\llvm\22.1.3\bin\clang
checking for the c compiler (cc) ... clang
checking for D:\scoop\apps\llvm\22.1.3\bin\clang ... d:\scoop\apps\llvm\22.1.3\bin\clang.exe
checking for flags (-fdiagnostics-color=always) ... ok
checking for flags (-fansi-escape-codes) ... ok
checking for clang++ ... D:\scoop\apps\llvm\22.1.3\bin\clang++
checking for the c++ compiler (cxx) ... clang++
checking for flags (-Wno-gnu-line-marker -Werror) ... ok
checking for clang++ ... D:\scoop\apps\llvm\22.1.3\bin\clang++
checking for the linker (ld) ... clang++
lld-link: error: could not open 'cpp-httplib.lib': no such file or directory
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> checking for c links(cpp-httplib)
> checking for c snippet(find_package/cpp-httplib)
checking for cpp-httplib ... no
note: install or modify (m) these packages (pass -y to skip confirm)?
in xmake-repo:
  -> cpp-httplib v0.41.0 [ssl:y, license:MIT]
please input: y (y/n/m)

checking for curl ... D:\scoop\apps\xmake\current\winenv\bin\curl
pinging the host(github.com) ... 259 ms
checking for D:\scoop\apps\llvm\22.1.3\bin\clang++ ... d:\scoop\apps\llvm\22.1.3\bin\clang++.exe
checking for flags (-fdiagnostics-color=always) ... ok
checking for flags (-Wno-gnu-line-marker -Werror) ... ok
> checking for c++ snippet(#define SIZE (si)
checking for tar ... C:\Windows\System32\tar.exe
C:\Windows\System32\tar.exe -xf F:\xmake\.xmake\cache\packages\2604\c\cpp-httplib\v0.41.0\cpp-httplib-v0.41.0.tar.gz
patching patches/v0.23.1/fix-mingw.diff to cpp-httplib-v0.41.0 ..
applying patch F:\xmake\.xmake\tmp\.xmake\260411\patches\cpp-httplib\v0.41.0\fix-mingw.diff
D:\scoop\apps\git\current\cmd\git.exe apply --reject --ignore-whitespace F:\xmake\.xmake\tmp\.xmake\260411\patches\cpp-httplib\v0.41.0\fix-mingw.diff
Checking patch httplib.h...
Hunk #1 succeeded at 5368 (offset 2311 lines).
Applied patch httplib.h cleanly.
checking for ninja ... D:\scoop\shims\ninja.exe
checking for cmake ... D:\scoop\shims\cmake.exe
checking for pkgconf ... no
checking for pkg-config ... no
checking for flags (clang_ms_runtime_lib) ... ok
checking for flags (clang_ms_runtime_lib) ... ok
checking for flags (clang_ms_runtime_lib) ... ok
checking for llvm-ar ... D:\scoop\apps\llvm\22.1.3\bin\llvm-ar
checking for the static library archiver (ar) ... llvm-ar
checking for clang++ ... D:\scoop\apps\llvm\22.1.3\bin\clang++
checking for the shared library linker (sh) ... clang++
checking for flags (clang_ms_runtime_lib) ... ok
checking for pkgconf ... no
checking for pkg-config ... no
D:\scoop\shims\cmake.exe -DHTTPLIB_COMPILE=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DHTTPLIB_REQUIRE_OPENSSL=ON -DHTTPLIB_REQUIRE_ZLIB=OFF -DHTTPLIB_REQUIRE_BROTLI=OFF -DHTTPLIB_NO_EXCEPTIONS=OFF -DOPENSSL_ROOT_DIR=F:\xmake\.xmake\packages\o\openssl3\3.6.1\0b71f9da82af473cbbb4aca35af3e7a0 -DCMAKE_INSTALL_PREFIX=F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810 -DCMAKE_INSTALL_LIBDIR:PATH=lib -G Ninja -DCMAKE_MAKE_PROGRAM=D:\scoop\shims\ninja.exe -DCMAKE_PDB_OUTPUT_DIRECTORY=F:/xmake/.xmake/cache/packages/2604/c/cpp-httplib/v0.41.0/source/build_611d6652/pdb -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL -DCMAKE_C_COMPILER=D:/scoop/apps/llvm/22.1.3/bin/clang.exe -DCMAKE_CXX_COMPILER=D:/scoop/apps/llvm/22.1.3/bin/clang++.exe -DCMAKE_COMPILE_PDB_OUTPUT_DIRECTORY=F:/xmake/.xmake/cache/packages/2604/c/cpp-httplib/v0.41.0/source/build_611d6652/pdb -DCMAKE_POLICY_VERSION_MINIMUM=3.5 -DCMAKE_STATIC_LINKER_FLAGS= "-DCMAKE_CXX_FLAGS= --target=x86_64-pc-windows-msvc" -DCMAKE_STATIC_LINKER_FLAGS_RELEASE= -DCMAKE_SHARED_LINKER_FLAGS= "-DCMAKE_SHARED_LINKER_FLAGS_RELEASE= -resource-dir=D:\\scoop\\apps\\llvm\\22.1.3\\lib\\clang\\22 -LD:\\scoop\\apps\\llvm\\22.1.3\\lib\\clang\\22\\lib -nostdlib" "-DCMAKE_CXX_FLAGS_RELEASE=-O3 -DNDEBUG -Xclang --dependent-lib=windows\\clang_rt.builtins-x86_64.lib -fms-runtime-lib=dll" "-DCMAKE_EXE_LINKER_FLAGS_RELEASE= -resource-dir=D:\\scoop\\apps\\llvm\\22.1.3\\lib\\clang\\22 -LD:\\scoop\\apps\\llvm\\22.1.3\\lib\\clang\\22\\lib -nostdlib" "-DCMAKE_C_FLAGS_RELEASE=-O3 -DNDEBUG -Xclang --dependent-lib=windows\\clang_rt.builtins-x86_64.lib -fms-runtime-lib=dll" "-DCMAKE_C_FLAGS= --target=x86_64-pc-windows-msvc" "-DCMAKE_MODULE_LINKER_FLAGS_RELEASE=  -resource-dir=D:\\scoop\\apps\\llvm\\22.1.3\\lib\\clang\\22 -LD:\\scoop\\apps\\llvm\\22.1.3\\lib\\clang\\22\\lib -nostdlib" F:\xmake\.xmake\cache\packages\2604\c\cpp-httplib\v0.41.0\source
-- The CXX compiler identification is Clang 22.1.3 with GNU-like command-line
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/scoop/apps/llvm/22.1.3/bin/clang++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - no
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
-- Found OpenSSL: F:/xmake/.xmake/packages/o/openssl3/3.6.1/0b71f9da82af473cbbb4aca35af3e7a0/lib/libcrypto.lib (found suitable version "3.6.1", minimum required is "3.0.0") found components: Crypto SSL
-- Configuring done (6.4s)
-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY
    CMAKE_C_COMPILER
    CMAKE_C_FLAGS
    CMAKE_C_FLAGS_RELEASE
    CMAKE_PDB_OUTPUT_DIRECTORY


-- Build files have been written to: F:/xmake/.xmake/cache/packages/2604/c/cpp-httplib/v0.41.0/source/build_611d6652
checking for pkgconf ... no
checking for pkg-config ... no
D:\scoop\shims\ninja.exe install -C F:\xmake\.xmake\cache\packages\2604\c\cpp-httplib\v0.41.0\source\build_611d6652 -j 18
ninja: Entering directory `F:\xmake\.xmake\cache\packages\2604\c\cpp-httplib\v0.41.0\source\build_611d6652'
[0/1] Install the project...-- Install configuration: "Release"
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/include/httplib.h
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/lib/cmake/httplib/httplibConfig.cmake
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/lib/cmake/httplib/httplibConfigVersion.cmake
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/lib/cmake/httplib/FindBrotli.cmake
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/lib/cmake/httplib/httplibTargets.cmake
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/share/doc/httplib/README.md
-- Installing: F:/xmake/.xmake/packages/c/cpp-httplib/v0.41.0/9f1d390c528245a0bdc9b1c1f2efb810/share/licenses/httplib/LICENSE

checking for xmake::cpp-httplib ... cpp-httplib v0.41.0
{ 
  version = "v0.41.0",
  sysincludedirs = { 
    "F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810\include" 
  },
  license = "MIT",
  defines = "CPPHTTPLIB_OPENSSL_SUPPORT" 
}

patching F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810\lib\pkgconfig\cpp-httplib.pc ..
checking for flags (-Xclang --dependent-lib=windows\clang_rt.builtins-x86_64.lib -fms-runtime-lib=dll) ... ok
> checking for c++ includes(httplib.h)
> checking for c++ links(libssl, libcrypto, ws2_32, user32, crypt32, advapi32)
> checking for c++ snippet(test)
error: In file included from F:\xmake\.xmake\tmp\.xmake\260411\_7bb52eae1fc098aff02182e1066c233f.cpp:2:
In file included from F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810\include\httplib.h:304:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\algorithm:10:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_heap_algorithms.hpp:11:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:11:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_iter_core.hpp:10:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\utility:11:
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\type_traits:2469:22: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
 2469 | _NODISCARD constexpr auto _Float_abs_bits(const _Ty& _Xx) noexcept {
      |                      ^
In file included from F:\xmake\.xmake\tmp\.xmake\260411\_7bb52eae1fc098aff02182e1066c233f.cpp:2:
In file included from F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810\include\httplib.h:304:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\algorithm:10:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_heap_algorithms.hpp:11:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:11:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_iter_core.hpp:10:
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\utility:182:22: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
  182 | _NODISCARD constexpr auto&& _Tuple_get(tuple<_Types...>&& _Tuple) noexcept;
      |                      ^
In file included from F:\xmake\.xmake\tmp\.xmake\260411\_7bb52eae1fc098aff02182e1066c233f.cpp:2:
In file included from F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810\include\httplib.h:304:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\algorithm:10:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_heap_algorithms.hpp:11:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:11:
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_iter_core.hpp:509:5: error: statement not allowed in constexpr function
  509 |     for (; _Idx < _Count; ++_Idx) {
      |     ^
In file included from F:\xmake\.xmake\tmp\.xmake\260411\_7bb52eae1fc098aff02182e1066c233f.cpp:2:
In file included from F:\xmake\.xmake\packages\c\cpp-httplib\v0.41.0\9f1d390c528245a0bdc9b1c1f2efb810\include\httplib.h:304:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\algorithm:10:
In file included from D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\__msvc_heap_algorithms.hpp:11:
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:361:1: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
  361 | auto _Min_vectorized(_Ty* const _First, _Ty* const _Last) noexcept {
      | ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:366:16: error: cannot initialize return object of type 'int' with an rvalue of type 'void *'
  366 |         return reinterpret_cast<void*>(::__std_min_8u(_First, _Last));
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:404:1: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
  404 | auto _Max_vectorized(_Ty* const _First, _Ty* const _Last) noexcept {
      | ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:409:16: error: cannot initialize return object of type 'int' with an rvalue of type 'void *'
  409 |         return reinterpret_cast<void*>(::__std_max_8u(_First, _Last));
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:804:15: error: deduced return types are a C++14 extension
  804 |     constexpr decltype(auto) operator()(_Args&&... _Vals) noexcept(
      |               ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:817:22: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
  817 | _NODISCARD constexpr auto _Pass_fn(_Fn& _Func) noexcept {
      |                      ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1418:16: error: constexpr function's return type 'void' is not a literal type
 1418 | constexpr void _Adl_verify_range(const _Iter& _First, const _Sentinel& _Last) {
      |                ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1444:22: error: deduced return types are a C++14 extension
 1444 | _NODISCARD constexpr decltype(auto) _Get_unwrapped(_Iter&& _It)
      |                      ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1471:22: error: deduced return types are a C++14 extension
 1471 | _NODISCARD constexpr decltype(auto) _Get_unwrapped_unverified(_Iter&& _It) {
      |                      ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1509:22: error: deduced return types are a C++14 extension
 1509 | _NODISCARD constexpr decltype(auto) _Get_unwrapped_n(_Iter&& _It, const _Diff _Off) {
      |                      ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1543:16: error: constexpr function's return type 'void' is not a literal type
 1543 | constexpr void _Seek_wrapped(_Iter& _It, _UIter&& _UIt) {
      |                ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1616:22: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
 1616 | _NODISCARD constexpr auto _Idl_distance(const _Iter& _First, const _Iter& _Last) {
      |                      ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1623:16: error: no viable conversion from returned value of type '_Distance_unknown' to function return type 'int'
 1623 |         return _Distance_unknown{};
      |                ^~~~~~~~~~~~~~~~~~~
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1924:27: error: constexpr function's return type 'void' is not a literal type
 1924 |     friend constexpr void _Verify_range(
      |                           ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1930:20: error: constexpr function's return type 'void' is not a literal type
 1930 |     constexpr void _Verify_offset(const difference_type _Off) const noexcept {
      |                    ^
D:\Program Files\Microsoft Visual Studio\VC\Tools\MSVC\14.50.35717\include\xutility:1949:20: error: constexpr function's return type 'void' is not a literal type
 1949 |     constexpr void _Seek_to(const reverse_iterator<_Src>& _It) noexcept(noexcept(current._Seek_to(_It.current))) {
      |                    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
  => install cpp-httplib v0.41.0 .. failed
error: install failed!
```

--- a/packages/c/cpp-httplib/xmake.lua
+++ b/packages/c/cpp-httplib/xmake.lua
@@ -115,9 +115,13 @@ package("cpp-httplib")
     end)
 
     on_test(function (package)
+        local languages = "c++11"
+        if package:is_plat("windows") and package:has_tool("cxx", "clang", "clangxx") then
+            languages = "c++14"
+        end
         assert(package:check_cxxsnippets({test = [[
             void test() {
                 httplib::Client cli("http://cpp-httplib-server.yhirose.repl.co");
             }
-        ]]}, {configs = {languages = "c++11"}, includes = "httplib.h"}))
+        ]]}, {configs = {languages = languages}, includes = "httplib.h"}))
     end)

